src/main/java/nextstep/subway/entity/Sections.java

```diff
@@ -96,18 +112,14 @@ private void addSectionToMiddle(Section section) {
         Station newDownStation = section.getDownStation();
         Long newDistance = section.getDistance();
 
-        int oldIndex = 0;
-        Section oldSection = new Section();
-        for (int index = 0; index < sectionList.size(); index++) {
-            Section currentSection = sectionList.get(index);
-            if (currentSection.getUpStation().getId() == newUpStation.getId()) {
-                validateStationDuplication(newDownStation);
-                validateDistance(currentSection, newDistance);
-                oldIndex = index;
-                oldSection = currentSection;
-                break;
-            }
+        Optional<Section> sectionByUpStation = findSectionByUpStation(section.getUpStation());
+        if (!sectionByUpStation.isPresent()) {
+            throw new IllegalSectionException("구간을 추가할 수 없습니다.");
```

Comment:
```suggestion
        Section sectionByUpStation = findSectionByUpStation(section.getUpStation()).orElseThrow(...);
```
이러면 고민하신부분들이 모두 해결되지 않을까 싶네요!